defold-spine/src/comp_spine_model.cpp

```diff
@@ -727,25 +734,32 @@ namespace dmSpine
             }
 
             component.m_DoRender = 1;
+
+            // Update bounding boxes
+            SpineModelBounds& bounds = world->m_BoundingBoxes[i];
```

Comment:
Well, yes, the m_Objects is an object pool, not an array.
See `comp_spine_model.cpp`
The items are returned to the pool when a component is destroyed.
This is going to happen during the frame (I wouldn't assume too much on _when_ it happens)

The `count` here doesn't actually represent the indices into the object pool.
Think of the pool as a sparse array.
You'll see the pool `index` in the `CompSpineModelCreate`

We currently don't store the pool index (we haven't needed to).
You could use the `count` variable for generating indices, if you can guarantuee that you'll use the same index later on.
I am note sure we can do that though, as not all items are sent to the render queue, thus not all of them are going to get the same index. Getting updated might not mean the same as being rendered.

So, you need to make sure the correct bounding volume is stored with the render item, so that you can match them correctly.
See how `comp_sprite.cpp` does it for reference.
It works because the `CalcBoundingVolumes` and render item loop iterates the same way. thus keeping the indices matching.